### PR TITLE
INV-17 Text based query

### PR DESCRIPTION
**Pull request by gordocanchola**
starting the http transport when the app comes up¬
  wrapping the query reducers in a 'query.reducers' key¬
  adding the 'query/SOURCE-ERROR' action type to handle errors returned from            querying the backend¬
  updating the transport to handle a filterTree input in addition to the cql string¬
  adding the `query.reducers` to the redux store¬
  creating the BasicSearch component¬
    currently only a text search⋅¬
    connected to redux to dispatch the executeQuery action¬
  adding the BasicSearch component to the routes¬
  adding storybook story for BasicSearch

### CODE REVIEW

File: src/main/webapp/basic-search.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          const value = e.target.value
+          setFilterTree({ ...filterTree, value })
+        }}
+      />
+
+      <Button
+        fullWidth
+        variant="contained"
+        color="primary"
+        onClick={() => {
```
Comment (Bdthomson) [inline src/main/webapp/basic-search.js]: ✏️ Could be one line
```suggestion
        onClick={() => props.onSearch(filterTree)}

```

File: src/main/webapp/basic-search.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          props.onSearch(filterTree)
+        }}
+      >
+        Search
+      </Button>
+    </CardContent>
+  )
+}
+
+const populateDefaultFilterTree = filterTree => ({
```
Comment (Bdthomson) [inline src/main/webapp/basic-search.js]: Do we want this default data committed?

File: src/main/webapp/intrigue-api/lib/cache.js [hunk 3]
```diff
@@ -142,37 +162,42 @@ const getSourceStatus = queryId => state => {
...
 }
 
 const getMetacards = state => {
-  return state.results.toJSON()
+  return getLocalState(state).results.toJSON()
 }
 
 const getMetacard = metacardId => state => {
-  const metacard = state.results.get(metacardId)
+  const metacard = getLocalState(state).results.get(metacardId)
```
Comment (Bdthomson) [inline src/main/webapp/intrigue-api/lib/cache.js]: What is this get local state pattern?

File: src/main/webapp/basic-search.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  id: '313a84858daa4ef5980d4b11a745d6d3',
+  spellcheck: false,
+  phonetics: false,
+  batchId: '5a3f400c2e1e410e8d37494500173ca4',
+})
+
+export const BasicSearch = props => {
+  return (
+    <div style={{ maxWidth: 600, margin: '20px auto' }}>
+      <Card>
```
Comment (adimka) [inline src/main/webapp/basic-search.js]: We should have a discussion on whether we want to use Card or Paper as our core building block.

File: src/main/webapp/basic-search.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      >
+        Search
+      </Button>
+    </CardContent>
+  )
+}
+
+const populateDefaultFilterTree = filterTree => ({
+  // hello source to test error case
+  srcs: ['ddf.distribution', 'cache', 'hello'],
```
Comment (adimka) [inline src/main/webapp/basic-search.js]: Do we want to add Error cases in our actual code, or would this be better to do in a Story?

Comment (gordocanchola) [reply src/main/webapp/basic-search.js]: The error case was to test the error action and reducer and the Story isn't using the redux connected BasicSearch.
I'll remove it.

File: src/main/webapp/basic-search.js (hunk 1 above)
Comment (gordocanchola) [reply src/main/webapp/basic-search.js]: Good catch. 
Changed

File: src/main/webapp/intrigue-api/lib/cache.js (hunk 3 above)
Comment (gordocanchola) [reply src/main/webapp/intrigue-api/lib/cache.js]: We were just trying to group the query reducers under a unique key. 
The get local state is just pulling that key from state

File: src/main/webapp/basic-search.js (hunk 2 above)
Comment (gordocanchola) [reply src/main/webapp/basic-search.js]: We do want to commit this so the search will work.
The idea is that we'll populate the query with the real values as the components are added.

Renamed to `populateDefaultQuery`

**Pull request merged**